nbclient/util.py

```diff
@@ -45,3 +47,17 @@ def wrapped(self, *args, **kwargs):
         return result
     wrapped.__doc__ = coro.__doc__
     return wrapped
+
+
+async def await_or_block(func, *args, **kwargs):
+    """Awaits the function if it's an asynchronous function. Otherwise block
+    on execution.
+    """
+    if asyncio.iscoroutinefunction(func):
+        return await func(*args, **kwargs)
+    else:
+        result = func(*args, **kwargs)
+        # Mocks mask that the function is a coroutine :/
+        if isinstance(result, Coroutine):
+            return await result
+        return result
```

Comment:
In jupyter_server, we will have something similar that we call [ensure_async](https://github.com/jupyter/jupyter_server/blob/a2610d33be535b42e37e3c200bcec6afeec341d8/jupyter_server/utils.py#L386-L400). We use it as a wrapper on a function call when we don't know it this function is async or not. I prefer that over passing the function object and its parameters to `await_or_block` and let it make the call. I will try it and maybe make a PR on your `zombieProcFix` branch.

Reply:
I'm ok with that change, thanks for helping